LDAR_Sim/src/programs/method.py

```diff
@@ -439,27 +439,27 @@ def check_weather(self, weather, curr_date, site: Site) -> bool:
         PREICIP = "precip"
 
         # TODO change to getters
-        lat = site._lat
-        long = site._long
+        lat = np.where(weather.latitude == math.floor(site._lat))[0][0]
+        long = np.where(weather.longitude == math.floor(site._long))[0][0]
         bool_temp: bool = False
         bool_wind: bool = False
         bool_precip: bool = False
 
         if (
             self._weather_envs[TEMP][0]
-            <= weather.temps[curr_date, lat, long]
+            <= weather.temps[curr_date.timetuple().tm_yday, lat, long]
```

Comment:
There should be an easier way to get this value from a Date object than converting to a timetuple - It seems like something that should be accessible as a property

Reply:
Another approach is to subtract the current date from the start of the year date. However, from what I can find, the performance difference between this method and using `timetuple()` appears to be quite negligible.